integration/qbft/scenarios/regular.go

```diff
@@ -168,51 +173,36 @@ func regularInstanceValidator(consensusData *spectypes.ConsensusData, operatorID
 			},
 		}
 
-		expectedStateRoot, err := expectedStoredInstance.State.GetRoot()
-		if err != nil {
-			return fmt.Errorf("error during geting root from expected state: %w", err)
-		}
-
-		actualStateRoot, err := actual.State.GetRoot()
-		if err != nil {
-			return fmt.Errorf("error during geting root from actual state: %w", err)
+		if err := validateByRoot(expected.State, actual.State); err != nil {
+			return err
 		}
 
-		if !bytes.Equal(expectedStateRoot, actualStateRoot) {
-			return fmt.Errorf("expected and actual state roots doesn't matching, expected state = %+v, actual = %+v", expectedStoredInstance.State, actual.State)
-		}
-
-		expectedDecidedMessageRoot, err := expectedStoredInstance.DecidedMessage.GetRoot()
-		if err != nil {
-			return fmt.Errorf("error during geting root from expected decided message: %w", err)
+		if err := validateByRoot(expected.DecidedMessage, actual.DecidedMessage); err != nil {
+			return err
 		}
 
-		actualDecidedMessageRoot, err := actual.DecidedMessage.GetRoot()
-		if err != nil {
-			return fmt.Errorf("error during geting root from actual decided message: %w", err)
-		}
+		return nil
+	}
+}
 
-		if !bytes.Equal(expectedDecidedMessageRoot, actualDecidedMessageRoot) {
-			return fmt.Errorf("expected and actual decided message roots doesn't matching, expected decided message = %+v, actual = %+v", expectedStoredInstance.DecidedMessage, actual.DecidedMessage)
+func validateSignedMessage(expected, actual *specqbft.SignedMessage) error {
+	for i := range expected.Signers {
+		if expected.Signers[i] != actual.Signers[i] {
+			return fmt.Errorf("signers not matching. expected = %+v, actual = %+v", expected.Signers, actual.Signers)
 		}
+	}
 
-		return nil
+	if err := validateByRoot(expected, actual); err != nil &&
+		expected.Signers[0] == actual.Signers[0] {
```

Comment:
Do we need the second check? If `Signers[0]` are different, will the check above return an error?